src/test/regress/sql/propagate_foreign_servers.sql

```diff
@@ -0,0 +1,79 @@
+
+SELECT 1 FROM citus_add_node('localhost', :worker_1_port);
+
+-- verify the dependent schema and the foreign server are created on the newly added worker
+SELECT run_command_on_workers(
+        $$SELECT COUNT(*) FROM pg_namespace WHERE nspname = 'test_dependent_schema';$$);
+SELECT run_command_on_workers(
+        $$SELECT COUNT(*)=1 FROM pg_foreign_server WHERE srvname = 'foreign_server_dependent_schema';$$);
+
+CREATE SERVER foreign_server_to_drop
+        FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host 'test');
+
+--should error
+DROP SERVER foreign_server_dependent_schema, foreign_server_to_drop;
+
+SET client_min_messages TO ERROR;
+DROP SCHEMA test_dependent_schema CASCADE;
+RESET client_min_messages;
+
+-- test propagating foreign server creation
+CREATE EXTENSION postgres_fdw;
+CREATE SERVER foreign_server TYPE 'test_type' VERSION 'v1'
+        FOREIGN DATA WRAPPER postgres_fdw
+        OPTIONS (host 'testhost', port '5432', dbname 'testdb');
+
+SELECT COUNT(*)=1 FROM pg_foreign_server WHERE srvname = 'foreign_server';
+
+-- verify that the server is created on the worker
+SELECT run_command_on_workers(
+        $$SELECT COUNT(*)=1 FROM pg_foreign_server WHERE srvname = 'foreign_server';$$);
+
+ALTER SERVER foreign_server OPTIONS (ADD "fdw_startup_cost" '1000');
+ALTER SERVER foreign_server OPTIONS (ADD passfile 'to_be_dropped');
+ALTER SERVER foreign_server OPTIONS (SET host 'localhost');
+ALTER SERVER foreign_server OPTIONS (DROP port, DROP dbname);
+ALTER SERVER foreign_server OPTIONS (ADD port :'master_port', dbname 'regression', DROP passfile);
+ALTER SERVER foreign_server RENAME TO "foreign'server_1!";
+
+-- test alter owner
+SELECT rolname FROM pg_roles JOIN pg_foreign_server ON (pg_roles.oid=pg_foreign_server.srvowner) WHERE srvname = 'foreign''server_1!';
+ALTER SERVER "foreign'server_1!" OWNER TO pg_monitor;
+
+-- verify that the server is renamed on the worker
+SELECT run_command_on_workers(
+        $$SELECT srvoptions FROM pg_foreign_server WHERE srvname = 'foreign''server_1!';$$);
+
+-- verify the owner is changed
+SELECT run_command_on_workers(
+        $$SELECT rolname FROM pg_roles WHERE oid IN (SELECT srvowner FROM pg_foreign_server WHERE srvname = 'foreign''server_1!');$$);
+
+-- verify the owner is changed on the coordinator
+SELECT rolname FROM pg_roles JOIN pg_foreign_server ON (pg_roles.oid=pg_foreign_server.srvowner) WHERE srvname = 'foreign''server_1!';
+DROP SERVER IF EXISTS "foreign'server_1!" CASCADE;
+
+-- verify that the server is dropped on the worker
+SELECT run_command_on_workers(
+        $$SELECT COUNT(*)=0 FROM pg_foreign_server WHERE srvname = 'foreign''server_1!';$$);
+
+\c - - - :worker_1_port
+-- not allowed on the worker
+ALTER SERVER foreign_server OPTIONS (ADD async_capable 'False');
+
+CREATE SERVER foreign_server_1 TYPE 'test_type' VERSION 'v1'
+         FOREIGN DATA WRAPPER postgres_fdw
+         OPTIONS (host 'testhost', port '5432', dbname 'testdb');
+
+\c - - - :master_port
```

Comment:
I think we not leave any objects behind. So, maybe recreate test_dependent_schema and make sure once we drop, no objects left behind.

In general, left over objects can become painful once you do large refactors etc.